src/settings.rs

```diff
@@ -35,7 +20,10 @@ pub(crate) struct Rule {
 #[serde(default)]
 #[serde(rename_all = "camelCase")]
 pub(crate) struct Settings {
-    pub rules: Vec<Rule>,
+    pub contains_any_of: Option<HashSet<String>>,
+    pub does_not_contains_any_of: Option<HashSet<String>>,
+    pub contains_all_of: Option<HashSet<String>>,
+    pub does_not_contains_all_of: Option<HashSet<String>>,
 }
```

Comment:
We could simplify things (I'm looking at the validation function, lines 36-57) by leveraging Rust's type system.

I would turn `Settings` into an `enum` and I would use serde's [internal tagging of enums](https://serde.rs/enum-representations.html#internally-tagged).

The settings would change to be something like:

```yaml
settings:
   criteria: containsAnyOf # doesNotContainAnyOf | containsAllOf | doesNotContainAllOf
   vars:
   - a
   - b
```

That would become something like that:

```rust
#[derive(Serialize, Deserialize)]
#[serde(tag = "criteria")]
enum Settings {
  ContainsAnyOf { vars: HashSet<String> }
  DoesNotContainAnyOf { vars: HashSet<String> }
  // and so on
}
```

As a validation we would only need to make sure `!vars.empty()` and the regexp.